tutorial/README.md

```diff
@@ -0,0 +1,824 @@
+
+### 2. Using the db Component
+
+The database in this example is run as a normal `ContainerizedWorkload`. You can find the complete code in `./flight-tracker/db.arm`. This step will require you to understand the new features used by this component, and include it in your `ApplicationConfiguration`.
+
+Here are the contents of the `db` `Component` with some annotations to introduce the parameter concept. Most of this definition should look familiar from the examples with the hello-world application.
+
+```js
+// input statements declare parameters that can be set by the ApplicationConfiguration
+//
+// Use parameters to avoid hardcoding of environment-specific configuration. In this
+// the database username, database name, and database password will be passed in. Since
+// we're running the database in a container, these values will be used to configure
+// the database itself. Components that need to connect to the database will need the
+// same values to make a connection.
+//
+// Most parameters will be of type `string`. Use `securestring` when the contents are
+// sensitive.
+input string dbuser
+input string dbname
+input securestring dbpassword
+
+resource oam 'Component@core.oam.dev/v1alpha2' _ {
+  metadata: {
+    name: 'db'
+  }
+  spec : {
+    workload: {
+      apiVersion: 'core.oam.dev/v1alpha2'
+      kind: 'ContainerizedWorkload'
+      metadata: {
+        name: 'db'
+      }
+      spec: {
+        osType: 'linux'
+        arch: 'amd64'
+        containers: [
+          {
+            name: 'postgres'
+            image: 'docker.io/postgres:9.6.17-alpine'
+
+            // The `env` property specifies values for environment variables of the container.
+            // It's typical to use `env` with parameters to pass data from the ApplicationConfiguration
+            // into the container.
+            //
+            // The `env` property contains an array of objects with the `name` and `value` properties set.
+            env: [
+              {
+                // Referencing the value of a parameter doesn't require a special syntax, just use the parameter name.
+                name: 'POSTGRES_USER'
+                value: dbuser
+              }
+              {
+                name: 'POSTGRES_PASSWORD'
+                value: dbpassword
+              }
+              {
+                name: 'POSTGRES_DB'
+                value: dbname
+              }
+            ]
+            ports: [
+              {
+                name: 'postgres'
+                containerPort: 5432
+                protocol: 'TCP'
+              }
+            ]
+          }
+        ]
+      }
+    }
+  }
+}
+```
+
+The `db` `Component` is already defined for you in `./flight-tracker/db.arm`.
+
+Now that's include it in the (currently empty `ApplicationConfiguration`) you created in the previous step. It should look like the following:
+
+```js
+resource oam 'ApplicationConfiguration@core.oam.dev/v1alpha2' _ {
+  // ... metadata omitted
```

Comment:
Do not omit metadata (same everywhere) - it makes learning harder since readers might be asking themselves what should it be.